src/Functions/CastOverloadResolver.cpp

```diff
@@ -167,9 +167,143 @@ REGISTER_FUNCTION(CastOverloadResolvers)
     factory.registerFunction("_CAST", [](ContextPtr context){ return CastOverloadResolverImpl::create(context, CastType::nonAccurate, true, {}); }, {}, FunctionFactory::Case::Insensitive);
     /// Note: "internal" (not affected by null preserving setting) versions of accurate cast functions are unneeded.
 
-    factory.registerFunction("CAST", [](ContextPtr context){ return CastOverloadResolverImpl::create(context, CastType::nonAccurate, false, {}); }, {}, FunctionFactory::Case::Insensitive);
-    factory.registerFunction("accurateCast", [](ContextPtr context){ return CastOverloadResolverImpl::create(context, CastType::accurate, false, {}); }, {});
-    factory.registerFunction("accurateCastOrNull", [](ContextPtr context){ return CastOverloadResolverImpl::create(context, CastType::accurateOrNull, false, {}); }, {});
+    /// CAST documentation
+    FunctionDocumentation::Description CAST_description = R"(
+Converts a value to a specified data type. Unlike conversion functions, `CAST` allows conversions between a broader set of data type pairs, including potentially unsafe conversions that may result in data loss or exceptions.
+
+The cast can be performed to any target type.
+    )";
+    FunctionDocumentation::Syntax CAST_syntax = R"(
+CAST(x, T) | CAST(x AS T) | x::T
```

Comment:
I like `|` but something like this will be easier to read for the general audience:

```
CAST(x, T) or CAST(x AS T) or x::T
```